src/main/java/org/example/springboot/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.example.springboot.repository;
+
+import java.util.List;
+import org.example.springboot.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Unable to save book to DB: " + book);
```

Comment:
on DAO layer use `DataProcessingException`
```suggestion
            throw new DataProcessingException("Unable to save book to DB: " + book);
```